to.etc.domui/src/main/java/to/etc/domui/converter/NumberConverterFactory.java

```diff
@@ -48,4 +57,23 @@ public int accept(Class< ? > clz, PropertyMetaModel< ? > pmm) {
 	public <X, T extends IConverter<X>> T createConverter(Class<X> clz, PropertyMetaModel<X> pmm) {
 		return (T) new NumberConverter<>((Class<? extends Number>)clz, pmm.getNumericPresentation(), pmm.getScale() < 0 ? 0 : pmm.getScale());
 	}
+
+	/**
+	 * Returns reusable {@link BigDecimalNumberConverter} instance based on specified numeric presentation and scale.
+	 * @return
+	 */
+	static public IConverter<BigDecimal> createBigDecimalNumberConverters(NumericPresentation np, int scale) {
+		Pair<NumericPresentation, Integer> key = new Pair<>(np, Integer.valueOf(scale));
+		NumberConverter<BigDecimal> bdConverter = BIG_DECIMAL_CONVERTERS.get(key);
+		if (null == bdConverter) {
```

Comment:
You cannot do this; this is a common pattern that fails. The best way to handle this is to use a ConcurrentHashMap, and to use computeIfAbsent - that will atomically do the right thing.